sources/packages/api/src/route-controllers/designation-agreement/designation-agreement.aest.controller.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  Controller,
+  Get,
+  Param,
+  NotFoundException,
+  Patch,
+  Body,
+  Query,
+  BadRequestException,
+} from "@nestjs/common";
+import { DesignationAgreementService, FormService } from "../../services";
+import { FormNames } from "../../services/form/constants";
+import { DesignationAgreementStatus } from "../../database/entities";
+import { getISODateOnlyString, PaginationParams } from "../../utilities";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, Groups, UserToken } from "../../auth/decorators";
+import { IUserToken } from "../../auth/userToken.interface";
+import { UserGroups } from "../../auth/user-groups.enum";
+import {
+  GetDesignationAgreementDto,
+  GetDesignationAgreementsDto,
+  PendingDesignationDto,
+  UpdateDesignationDto,
+} from "./models/designation-agreement.model";
+import { DesignationAgreementServiceController } from "./designation-agreement.service.controller";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("designation-agreement")
+export class DesignationAgreementAESTController {
+  constructor(
+    private readonly designationAgreementServiceController: DesignationAgreementServiceController,
+    private readonly designationAgreementService: DesignationAgreementService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Retrieve the designation agreement information and
+   * the associated locations approvals.
+   * @param designationId
+   * @returns designation agreement information.
+   */
+  @Get(":designationId")
+  async getDesignationAgreement(
+    @Param("designationId") designationId: number,
+  ): Promise<GetDesignationAgreementDto> {
+    return this.designationAgreementServiceController.getDesignationAgreement(
+      designationId,
+    );
+  }
+
+  /**
+   * Get the list of all the designations that belongs to
+   * the institution.
+   * @param institutionId
+   * @returns the list of all the designations that
+   * belongs to the institution.
+   */
+  @Get("institution/:institutionId")
+  async getDesignationAgreements(
+    @Param("institutionId") institutionId: number,
+  ): Promise<GetDesignationAgreementsDto[]> {
+    return this.designationAgreementServiceController.getDesignationAgreements(
+      institutionId,
+    );
+  }
+
+  /**
+   * API to retrieve all pending designations.
+   * @param designationStatus
+   * @param searchCriteria to search designation.
+   * @returns Pending designations.
+   */
+  @Get("status/:designationStatus")
+  async getDesignationAgreementByStatus(
+    @Param("designationStatus") designationStatus: DesignationAgreementStatus,
+    @Query(PaginationParams.SearchCriteria) searchCriteria: string,
+  ): Promise<PendingDesignationDto[]> {
+    if (
+      !Object.values(DesignationAgreementStatus).includes(designationStatus)
```

Comment:
i believe, here` DesignationAgreementStatus` is Enum, (type of `designationStatus`) i.e 
![image](https://user-images.githubusercontent.com/77353155/156650150-622d19da-b651-4c45-9d72-bb741bff7b86.png)
I am not understanding this logic. could you explain or am I missing anything?

Reply:
This API retrieves designation agreements by designation status(API built in generic way but to get pending designations) so designationStatus is a path param here. Line 80 validates if the path param consists only acceptable values from enum.

Reply:
@ann-aot, this will ensure that the string passed as a parameter actually belongs to the enum.
It it is not clear at a first sight we can encapsulate this in a utils method with a meaningful name.